### PR TITLE
Added code to fix broken integration tests

### PR DESCRIPTION
**Pull request by pramireddy**
Added code to fix broken integration tests

### CODE REVIEW

File: DFC.App.Banners.IntegrationTests/CustomWebApplicationFactory.cs [hunk 1]
```diff
@@ -1,13 +1,52 @@
...
     {
+        public CustomWebApplicationFactory()
+        {
+            this.MockCosmosRepo = A.Fake<ICosmosRepository<PageBannerContentItemModel>>();
+        }
+
+        internal ICosmosRepository<PageBannerContentItemModel> MockCosmosRepo { get; set; }
+
+        public HttpClient CreateClientWithWebHostBuilder()
+        {
```
Comment (ravi-methods) [inline DFC.App.Banners.IntegrationTests/CustomWebApplicationFactory.cs:26]: This is probably not needed, `builder.RegisterServices(this.MockCosmosRepo);` can move to `protected override void ConfigureWebHost(IWebHostBuilder builder)` method and you can keep using `var client = this.factory.CreateClient();` in your test classes.

Comment (pramireddy) [reply DFC.App.Banners.IntegrationTests/CustomWebApplicationFactory.cs:26]: We can do that way as well. But current implementation gives more flexible and allows to use orginal statup (without mocks) with 'CreateClient' and startup with mocked classes with 'CreateClientWithWebHostBuilder'.

File: DFC.App.Banners.IntegrationTests/ControllerTests/BannersControllerRouteTests.cs [hunk 2]
```diff
@@ -27,18 +25,14 @@ public BannersControllerRouteTests(CustomWebApplicationFactory<Startup> factory)
             new object[] { "/banners/document/reers-advice/career-choices-at-16-test" },
             new object[] { "/banners/document/action-plans-test" },
             new object[] { "/banners/document/contact-us-test" },
-            new object[] { "/banners/body/careers-advice-test" },
```
[the remark this answers is not in the document]
Comment (pramireddy) [reply DFC.App.Banners.IntegrationTests/ControllerTests/BannersControllerRouteTests.cs]: Update the code to include tests for "/banners/body/*..'

File: DFC.App.Banners.IntegrationTests/ControllerTests/HealthControllerRouteTests.cs [hunk 3]
```diff
@@ -34,12 +35,13 @@ public async Task GetHealthHtmlContentEndpointsReturnSuccessAndCorrectContentTyp
...
             var uri = new Uri(path, UriKind.Relative);
-            var client = this.factory.CreateClient();
+            var client = this.factory.CreateClientWithWebHostBuilder();
             client.DefaultRequestHeaders.Accept.Clear();
             client.DefaultRequestHeaders.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue(MediaTypeNames.Text.Html));
+            A.CallTo(() => this.factory.MockCosmosRepo.PingAsync()).Returns(true);
 
             // Act
-            var response = await client.GetAsync(uri);
+            var response = await client.GetAsync(uri).ConfigureAwait(false);
```
[the remark this answers is not in the document]
Comment (pramireddy) [reply DFC.App.Banners.IntegrationTests/ControllerTests/HealthControllerRouteTests.cs]: Updated code to remove '.ConfigureAwait(false)'

File: DFC.App.Banners.IntegrationTests/CustomWebApplicationFactory.cs (hunk 1 above)
Comment (ravi-methods) [reply DFC.App.Banners.IntegrationTests/CustomWebApplicationFactory.cs:26]: but we'd only mock the external services, and those services will always be mocked in integration tests, so I don't see a need for that flexibility.

Comment (pramireddy) [reply DFC.App.Banners.IntegrationTests/CustomWebApplicationFactory.cs:26]: that's fine. I am happy to update as to suggested.

Comment (pramireddy) [reply DFC.App.Banners.IntegrationTests/CustomWebApplicationFactory.cs]: Updated code to resolve PR comments

**Review by ravi-methods — APPROVED** [review]

**Pull request merged**